### PR TITLE
create a respondWithErrors and setDefaultErrorsResponse for errors

### PR DESCRIPTION
**Pull request by bhargavraviya**
we need multi errors showing function in API response that why create a new respondWithErrors and setDefaultErrorsResponse for errors

@ultrono and @f9web

### CODE REVIEW

**Comment by ultrono** [comment]
Thanks for this. 

Can you please add a test to ensure the current functionality of this method does not break and your new functionality works as expected. I suspect the test would look very similar to https://github.com/f9webltd/laravel-api-response-helpers/blob/master/tests/ResponseTest.php#L44-L55

**Comment by bhargavraviya** [comment]
> Thanks for this.
> 
> Can you please add a test to ensure the current functionality of this method does not break and your new functionality works as expected. I suspect the test would look very similar to https://github.com/f9webltd/laravel-api-response-helpers/blob/master/tests/ResponseTest.php#L44-L55

thank guide me, I have created a test case for respondWithErrors

File: src/ApiResponseHelpers.php [hunk 1]
```diff
@@ -81,6 +82,34 @@ public function respondError(?string $message = null): JsonResponse
...
+
+        return $this->apiResponse(
+            ['errors' => $data],
+            Response::HTTP_BAD_REQUEST
+        );
+        $data = [] === $contents
+        ? $this->_api_helpers_defaultSuccessData
+        : $contents;
+
+    return $this->apiResponse($data);
```
Comment (ultrono) [inline src/ApiResponseHelpers.php]: Are these lines here in error, as you appear to be returning on the line #99.

**Comment by ultrono** [comment]
Looks good, have added a couple of comments to your code.

**Comment by ultrono** [comment]
@bhargavraviya is this a feature you're still planning?

**Comment by bhargavraviya** [comment]
> @bhargavraviya is this a feature you're still planning?

Yes but given some error on the test case. But I don't know how to fix it ?

Plz guide me

**Comment by ultrono** [comment]
> 

Sure, which error is the issue? I've added comments to the problem areas.

**Comment by bhargavraviya** [comment]
It shows "Merging is blocked" "Merging can be performed automatically with 1 approving review."
It automatically generates an error on merge by bot check

**Comment by ultrono** [comment]
> 

That's because the code you've written causes all automated tests to fail. If you click the "details" you can see exact reasons i.e. https://github.com/f9webltd/laravel-api-response-helpers/actions/runs/4729011609/jobs/8394078674?pr=27

**Comment by bhargavraviya** [comment]
Hello, my friend
Please help with the code.
Why this test case is broken
https://github.com/f9webltd/laravel-api-response-helpers/compare/master...bhargavraviya:laravel-api-response-helpers:master

Also you have seen my whole code on my repository 
https://github.com/bhargavraviya/laravel-api-response-helpers

**Comment by ultrono** [comment]
> Hello, my friend Please help with the code. Why this test case is broken [master...bhargavraviya:laravel-api-response-helpers:master](https://github.com/f9webltd/laravel-api-response-helpers/compare/master...bhargavraviya:laravel-api-response-helpers:master)
> 
> Also you have seen my whole code on my repository https://github.com/bhargavraviya/laravel-api-response-helpers

Hi,

All the code you've changed should be visible at https://github.com/f9webltd/laravel-api-response-helpers/pull/27/files.

YOu can see why tests failed by clickin on the "details" tab under the checks section i.e. https://github.com/f9webltd/laravel-api-response-helpers/actions/runs/4729011607/jobs/8394078391

I've added comments previously to all your code so you can fix.

Many Thanks

**Comment by bhargavraviya** [comment]
@ultrono 

In Test case give error 
plz help me on code 
```
public function testErrorsResponseDefaults(?array $default, $args, int $code, array $data): void
    {
        $response = $this->service->setDefaultErrorsResponse($default)->respondWithErrors($args);
        self::assertInstanceOf(JsonResponse::class, $response);
        self::assertSame($code, $response->getStatusCode());
        self::assertSame($data, $response->getData(true));
        self::assertJsonStringEqualsJsonString(json_encode($data, JSON_THROW_ON_ERROR), $response->getContent());
    }
    ```
    
    which part is wrong?

**Comment by ultrono** [comment]
Did you resolve the original errors discussed above? What is the output of your failing test?

**Comment by bhargavraviya** [comment]
> Did you resolve the original errors discussed above? What is the output of your failing test?

yes this type of error 
```
public function testErrorsResponseDefaults(?array $default, $args, int $code, array $data): void
    {
        $response = $this->service->setDefaultErrorsResponse($default)->respondWithErrors($args);
        self::assertInstanceOf(JsonResponse::class, $response);
        self::assertSame($code, $response->getStatusCode());
        self::assertSame($data, $response->getData(true));
        self::assertJsonStringEqualsJsonString(json_encode($data, JSON_THROW_ON_ERROR), $response->getContent());
    }
  ```
    
    which part is wrong?

**Comment by ultrono** [comment]
@bhargavraviya what is the output of the failing test?

**Comment by bhargavraviya** [comment]
@ultrono 
this part

public function testErrorsResponseDefaults(?array $default, $args, int $code, array $data): void
    {
        $response = $this->service->setDefaultErrorsResponse($default)->respondWithErrors($args);
        self::assertInstanceOf(JsonResponse::class, $response);
        self::assertSame($code, $response->getStatusCode());
        self::assertSame($data, $response->getData(true));
        self::assertJsonStringEqualsJsonString(json_encode($data, JSON_THROW_ON_ERROR), $response->getContent());
    }

**Comment by ultrono** [comment]
> @ultrono this part
> 
> public function testErrorsResponseDefaults(?array $default, $args, int $code, array $data): void { $response = $this->service->setDefaultErrorsResponse($default)->respondWithErrors($args); self::assertInstanceOf(JsonResponse::class, $response); self::assertSame($code, $response->getStatusCode()); self::assertSame($data, $response->getData(true)); self::assertJsonStringEqualsJsonString(json_encode($data, JSON_THROW_ON_ERROR), $response->getContent()); }

That's the test itself, not the output as asked.

**Comment by ultrono** [comment]
@bhargavraviya a major package version (`2.0.0`) has just been released. This will probably cause merge conflicts within your proposed code.

**Comment by ultrono** [comment]
Closing due to inactivity.

**Pull request closed without merging**